psyneulink/core/llvm/codegen.py

```diff
@@ -293,9 +293,19 @@ def gen_composition_exec(ctx, composition, *, tags:frozenset):
 
         # Generate a while not 'end condition' loop
         builder.position_at_end(loop_condition)
+        is_finished_flags = {}
+        for idx, node in enumerate(composition.nodes):
+            try:
+                node_state = builder.gep(state, [ctx.int32_ty(0), ctx.int32_ty(0), ctx.int32_ty(idx)])
+                is_finished_flag_ptr = helpers.get_state_ptr(builder, node, node_state, "is_finished_flag")
```

Comment:
looks like  `get_state_ptr` is the only one that can raise an exception, pls move the rest to `else` block.